### ISSUE TITLE
Add support for Xiaomi Mijia 1C 

### ISSUE THREAD

**Issue by robertybob** [homebridge-xiaomi-roborock-vacuum/homebridge-xiaomi-roborock-vacuum#150]
Hey. I’ve recently got this vacuum up and running and would love to see it integrated into this plugin. Please let me know if there’s anyway I can help

**Comment by afharo** [comment]
What is the log you can see when you try running this plugin?

**Comment by robertybob** [comment]
> 
> 
> What is the log you can see when you try running this plugin?

Thanks for the reply! I've spent the last two hours figuring out how to get Python up and running on my Windows machine in order to install python-miio. The device discovery produced 0 results so I've had to use ibackupviewer to get the token via the Mi Home backup, I've got the token decrypted successfully too, so I'm keeping my fingers crossed it'll work with the plugin!
![image](https://user-images.githubusercontent.com/2610395/80280368-9270ca00-86fb-11ea-9611-8d913aaf46d5.png)

I found it curious that the 'Model' didn't match up with that I was expecting! But then I've Googled it and found it [listed here](https://github.com/rytilahti/python-miio/issues/627#issue-564688505) too, so I'm not the first to discover this lol

**Comment by robertybob** [comment]
> 
> 
> What is the log you can see when you try running this plugin?

Robot vacuum
        Model: dreame.vacuum.mc1808
        IP address: 192.168.1.237
        Token: 774d666238424b70326c6e6e4e557679
        MAC: 44:23:7C:EC:19:1B



First of all I got this:
[4/25/2020, 14:05:01] [Xiaomi Mijia 1C] ERR connect | miio.device, next try in 2 minutes | Error: Could not connect to device, handshake timeout

Then 2 mins later:

[4/25/2020, 14:07:14] [Xiaomi Mijia 1C] STA getDevice | Connected to: 192.168.1.237
[4/25/2020, 14:07:14] [Xiaomi Mijia 1C] STA getDevice | Model: dreame.vacuum.mc1808
[4/25/2020, 14:07:14] [Xiaomi Mijia 1C] STA getDevice | State: undefined
[4/25/2020, 14:07:14] [Xiaomi Mijia 1C] STA getDevice | FanSpeed: undefined
[4/25/2020, 14:07:14] [Xiaomi Mijia 1C] STA getDevice | BatteryLevel: undefined
[4/25/2020, 14:07:26] [Xiaomi Mijia 1C] ERR getSerialNumber | Failed getting the firmware version. Error: Call to device timed out
at Timeout.retry [as _onTimeout] (/usr/local/lib/node_modules/homebridge-xiaomi-roborock-vacuum/node_modules/miio-nicoh88/lib/network.js:493:18)
at listOnTimeout (internal/timers.js:549:17)
at processTimers (internal/timers.js:492:7) {
code: 'timeout'
}
[4/25/2020, 14:07:26] [Xiaomi Mijia 1C] ERR getDevice | get_serial_number | Error: Call to device timed out
[4/25/2020, 14:07:26] [Xiaomi Mijia 1C] INF getFirmware | dreame.vacuum.mc1808 | Firmwareversion is 3.5.8_1059
[4/25/2020, 14:07:26] [Xiaomi Mijia 1C] STA getDevice | Firmwareversion: 3.5.8_1059
[4/25/2020, 14:07:26] [Xiaomi Mijia 1C] INF changedCleaning | dreame.vacuum.mc1808 | Cleaning is OFF.
[4/25/2020, 14:07:26] [Xiaomi Mijia 1C] MON changedCharging | dreame.vacuum.mc1808 | ChargingState is now false
[4/25/2020, 14:07:26] [Xiaomi Mijia 1C] INF changedCharging | dreame.vacuum.mc1808 | Charging is cancelled
[4/25/2020, 14:07:26] [Xiaomi Mijia 1C] INF changedCharging | dreame.vacuum.mc1808 | Robot not anymore in dock.


It has showed up as a Fan in the Home app so seems to be working ok, I'll test it out properly and report back

**Comment by robertybob** [comment]
First of all I got this:
[4/25/2020, 14:05:01] [Xiaomi Mijia 1C] ERR connect | miio.device, next try in 2 minutes | Error: Could not connect to device, handshake timeout

Then 2 mins later:

[4/25/2020, 14:07:14] [Xiaomi Mijia 1C] STA getDevice | Connected to: 192.168.1.237
[4/25/2020, 14:07:14] [Xiaomi Mijia 1C] STA getDevice | Model: dreame.vacuum.mc1808
[4/25/2020, 14:07:14] [Xiaomi Mijia 1C] STA getDevice | State: undefined
[4/25/2020, 14:07:14] [Xiaomi Mijia 1C] STA getDevice | FanSpeed: undefined
[4/25/2020, 14:07:14] [Xiaomi Mijia 1C] STA getDevice | BatteryLevel: undefined
[4/25/2020, 14:07:26] [Xiaomi Mijia 1C] ERR getSerialNumber | Failed getting the firmware version. Error: Call to device timed out
at Timeout.retry [as _onTimeout] (/usr/local/lib/node_modules/homebridge-xiaomi-roborock-vacuum/node_modules/miio-nicoh88/lib/network.js:493:18)
at listOnTimeout (internal/timers.js:549:17)
at processTimers (internal/timers.js:492:7) {
code: 'timeout'
}
[4/25/2020, 14:07:26] [Xiaomi Mijia 1C] ERR getDevice | get_serial_number | Error: Call to device timed out
[4/25/2020, 14:07:26] [Xiaomi Mijia 1C] INF getFirmware | dreame.vacuum.mc1808 | Firmwareversion is 3.5.8_1059
[4/25/2020, 14:07:26] [Xiaomi Mijia 1C] STA getDevice | Firmwareversion: 3.5.8_1059
[4/25/2020, 14:07:26] [Xiaomi Mijia 1C] INF changedCleaning | dreame.vacuum.mc1808 | Cleaning is OFF.
[4/25/2020, 14:07:26] [Xiaomi Mijia 1C] MON changedCharging | dreame.vacuum.mc1808 | ChargingState is now false
[4/25/2020, 14:07:26] [Xiaomi Mijia 1C] INF changedCharging | dreame.vacuum.mc1808 | Charging is cancelled
[4/25/2020, 14:07:26] [Xiaomi Mijia 1C] INF changedCharging | dreame.vacuum.mc1808 | Robot not anymore in dock.

It has showed up as a Fan in the Home app so seems to be working ok, I'll test it out properly and report back

**Comment by robertybob** [comment]
![DD7FFE2F-FCC1-4704-9AF0-718DC459EF15](https://user-images.githubusercontent.com/2610395/80283527-1c2a9280-8710-11ea-8ce5-604c1fbc1812.png)
![13A30529-620E-4DBD-A914-63C3B86809E2](https://user-images.githubusercontent.com/2610395/80283531-1fbe1980-8710-11ea-8039-ef4d5b888acb.png)

These are the messages that appear in the log when I turn it on via the Home app

**Comment by robertybob** [comment]
(and the vacuum doesn’t do anything)

**Comment by robertybob** [comment]
![FD9AA4D5-D535-4CBF-82A4-008A9544CDC7](https://user-images.githubusercontent.com/2610395/80283652-2bf6a680-8711-11ea-9321-4c08e2ac80eb.png)

**Comment by rain82k** [comment]
I have the Mijia 2 but I think it could also work with the addition of the Mijia 1c. Or?

If the plugin could support the Mijia 2 it would be really cool

**Comment by robertybob** [comment]
@afharo is any of that helpful?

**Comment by afharo** [comment]
It is indeed! I'm only concerned that these models may have a different set of methods to call.

We actually had an initial discussion about them in this other issue: https://github.com/nicoh88/homebridge-xiaomi-roborock-vacuum/issues/100#issuecomment-581424913

**Comment by robertybob** [comment]
> 
> 
> It is indeed! I'm only concerned that these models may have a different set of methods to call.
> 
> We actually had an initial discussion about them in this other issue: [#100 (comment)](https://github.com/nicoh88/homebridge-xiaomi-roborock-vacuum/issues/100#issuecomment-581424913)

Ah, that doesn't sound promising  lol

**Comment by afharo** [comment]
it's doable, but it may take me a while... especially since I don't have the model. 
If anyone is willing to help on the development (or finding the methods to call on each case), it will be very welcomed :)

**Comment by robertybob** [comment]
> it's doable, but it may take me a while... especially since I don't have the model.
> If anyone is willing to help on the development (either coding and providing a Pull Request or finding the methods to call on each case), it will be very much welcomed :)

Yeah sorry, I’m not in any way knowledgeable about this kind of thing, but I’m more than happy to run any commands or produce any logs which could help someone figure out how to get this working :)

**Comment by bedrin** [comment]
Hi @robertybob ! Can you please try `next` branch?
I had the same issues with my Xiaowa E202-02 - it didn't start cleaning with similar error.

**Comment by robertybob** [comment]
> Hi @robertybob ! Can you please try `next` branch?
> I had the same issues with my Xiaowa E202-02 - it didn't start cleaning with similar error.

Hello! I will update in the morning and I will let you know if it works :) Thank you for thinking of me when looking into your own issue

**Comment by robertybob** [comment]
@bedrin I’m afraid I’m still getting errors. These are some of those errors:

![A855F19F-9251-4100-821F-063D5C16BC83](https://user-images.githubusercontent.com/2610395/81677960-39b66680-9449-11ea-8db6-78ea69b53508.png)
![A4E2895D-E9F2-49C3-8F95-9A4E0E39FB54](https://user-images.githubusercontent.com/2610395/81677984-3de28400-9449-11ea-85db-fdccef38622f.png)
![B7221ED9-EE83-42B7-947A-09C28E69036C](https://user-images.githubusercontent.com/2610395/81677991-3e7b1a80-9449-11ea-853b-c66f0cee9490.png)
![7C8E538B-E4EF-4C88-9D9E-82D86855571E](https://user-images.githubusercontent.com/2610395/81678001-3fac4780-9449-11ea-9dc5-612a3322486b.png)
![255DC9E8-ABD3-463A-BA72-46D5EDA4AA02](https://user-images.githubusercontent.com/2610395/81678006-4044de00-9449-11ea-9030-f5d62ad4c230.png)
![A70DB19D-029E-4B01-98A6-250E089065E7](https://user-images.githubusercontent.com/2610395/81678012-40dd7480-9449-11ea-90af-2025d6da4174.png)

**Comment by afharo** [comment]
> Thank you for thinking of me when looking into your own issue

@bedrin rocks!

**Comment by afharo** [comment]
I sincerely think the Mijia models have a different set of commands. We might need to create a "new" class of Vacuum for them in the miio repository: https://github.com/homebridge-xiaomi-roborock-vacuum/miio/blob/master/lib/devices/vacuum.js

I think the easiest way to implement it is to have a copy of that file, but modifying the commands to call and map it to the model in https://github.com/homebridge-xiaomi-roborock-vacuum/miio/blob/master/lib/models.js#L50

The list of specific commands for this model may be obtained from the python library: https://github.com/rytilahti/python-miio/issues/550

**Comment by afharo** [comment]
It may be easier to maintain if, at some point, we migrate the miio library (only the vacuum necessary stuff) to this plugin so we don't need to maintain two modules :)

**Comment by robertybob** [comment]
@afharo I appreciate what you’re saying, you’re doing this for free in your spare time so I’m not going to put pressure on you to resolve this  :)

**Comment by afharo** [comment]
Thank you for your words and understanding @robertybob 😍 

We can plan a quick win here:
Let's try to create a copy of the vacuum.js file in the miio library, but using https://github.com/rytilahti/python-miio/blob/20f915c9589fed55544a5417abe3fd3d9e12d08d/miio/viomivacuum.py to find the matching internal commands/properties to be read from the vacuum.

For example:
```
		// Define the batteryLevel property for monitoring battery
		this.defineProperty('battery', {
			name: 'batteryLevel'
		});
```

Should be something like
```
		// Define the batteryLevel property for monitoring battery
		this.defineProperty('battary_life', {
			name: 'batteryLevel'
		});
```
_`battary_life` obtained from https://github.com/rytilahti/python-miio/blob/20f915c9589fed55544a5417abe3fd3d9e12d08d/miio/viomivacuum.py#L83-L85

**Comment by robertybob** [comment]
> Thank you for your words and understanding @robertybob 😍
> 
> We can plan a quick win here:
> Let's try to create a copy of the vacuum.js file in the miio library, but using https://github.com/rytilahti/python-miio/blob/20f915c9589fed55544a5417abe3fd3d9e12d08d/miio/viomivacuum.py to find the matching internal commands/properties to be read from the vacuum.
> 
> For example:
> 
> ```
> 		// Define the batteryLevel property for monitoring battery
> 		this.defineProperty('battery', {
> 			name: 'batteryLevel'
> 		});
> ```
> 
> Should be something like
> 
> ```
> 		// Define the batteryLevel property for monitoring battery
> 		this.defineProperty('battary_life', {
> 			name: 'batteryLevel'
> 		});
> ```
> 
> _`battary_life` obtained from https://github.com/rytilahti/python-miio/blob/20f915c9589fed55544a5417abe3fd3d9e12d08d/miio/viomivacuum.py#L83-L85_

That sounds good! I’m a novice at this kind of thing I’m afraid. If there is anything I need to do, would you be able to create a step-by-step list of commands I need to run? You may need to pretend I’m a child trying to do it 😄

**Comment by afharo** [comment]
haha! Don't you worry. It was just me adding a few developer notes to know when we can tackle it :)

I'll try to find some time this weekend to prepare a version for you to test :)

**Comment by robertybob** [comment]
>  I'll try to find some time this weekend to prepare a version for you to test :)

😀👍🏻

**Comment by afharo** [comment]
Can you try installing `homebridge-xiaomi-roborock-vacuum@next`? It should include the initial implementation of added support to ViomiVacuum :)

**Comment by robertybob** [comment]
@afharo  I’m still getting errors and can’t get the vacuum to start, but I think they are different errors this time which I assume is a good thing! 

![1035489F-352C-4676-936F-714EDFB80117](https://user-images.githubusercontent.com/2610395/82212819-16892c80-990b-11ea-880c-71bfab1e7fea.png)
![8756A606-E66F-4DC2-A7A3-F08BC7043A81](https://user-images.githubusercontent.com/2610395/82212835-1a1cb380-990b-11ea-9bcb-e5150e3343fa.png)
![ADA6AF58-F72C-4AF0-A3D6-AA9C2D55AF5F](https://user-images.githubusercontent.com/2610395/82212840-1b4de080-990b-11ea-8134-4e0e875ebe81.png)

**Comment by afharo** [comment]
oh! ok! I'll take a second look :(
Thanks for testing!

**Comment by afharo** [comment]
Can you please try again with `homebridge-xiaomi-roborock-vacuum@0.11.0-0`?

**Comment by robertybob** [comment]
> Can you please try again with `homebridge-xiaomi-roborock-vacuum@0.11.0-0`?

Thanks! I haven’t uninstalled the existing branch, hope that’s ok (I presumed it would overwrite the files rather than create a new instance of it)

**Comment by afharo** [comment]
Do you mean the `@next`? Yeah... as long as you re-run the install, it will overwrite :)

**Comment by robertybob** [comment]
@afharo I'll run `sudo npm install -g homebridge-xiaomi-roborock-vacuum@0.11.0-0 --unsafe-perm` right away and let you know if it works!

**Comment by afharo** [comment]
Thank you!

**Comment by robertybob** [comment]
@afharo 

Installation has finished and I got an error saying 
` WARN homebridge-xiaomi-roborock-vacuum@0.11.0-0 requires a peer of homebridge@>=0.4.45 but none is installed. You must install peer dependencies yourself.` should I be worried?

**Comment by afharo** [comment]
I think when that log is shown, it means it's installed in the wrong directory.

**Comment by afharo** [comment]
Where/How did you install `homebridge`? And how did you install `homebridge-xiaomi-roborock-vacuum`?

**Comment by robertybob** [comment]
I ran `sudo npm install -g homebridge-xiaomi-roborock-vacuum@0.11.0-0 --unsafe-perm` from the commandline. Was I supposed to run `sudo npm install -g homebridge-xiaomi-roborock-vacuum@latest --unsafe-perm`?

**Comment by robertybob** [comment]
@afharo I've pasted the error messages here https://pastebin.com/sjyqwN8K  :)

**Comment by afharo** [comment]
I see `[5/22/2020, 19:39:35] [Xiaomi Mi Robot Vaccum 1st Generation] ERR getState | {"code":"timeout"}`

I think that's quite enlightening. I'll see if I can revisit it somehow.

**Comment by robertybob** [comment]
I've installed the latest version you've just pushed and now getting slightly different error messages

https://pastebin.com/gTLUdftR

**Comment by afharo** [comment]
I've published another version. Hopefully it's better this time...

re the logs:

```
[5/22/2020, 20:08:56] [Xiaomi Mi Robot Vaccum 1st Generation] ERR getSerialNumber | Failed getting the serial number. Error: Call to device timed out
    at Timeout.retry [as _onTimeout] (/usr/local/lib/node_modules/homebridge-xiaomi-roborock-vacuum/miio/lib/network.js:519:23)
    at listOnTimeout (internal/timers.js:549:17)
    at processTimers (internal/timers.js:492:7) {
  code: 'timeout'
}
```

Will still show up. I can't find the command to retrieve the serial number in that model.

This one is the key one:
```[5/22/2020, 20:09:09] [Xiaomi Mi Robot Vaccum 1st Generation] ERR getState | {"code":"timeout"}```

**Comment by afharo** [comment]
@robertybob Does that thumb up mean it works now? 😇

**Comment by robertybob** [comment]
@afharo Oh, no sorry it’s still not working. I had already updated to that new version you mentioned and that’s where those errors were coming from. Every 30 seconds the errors come up, so perhaps I should just uninstall for the time being? 

![226891B0-6960-4A83-8281-35BF728386D1](https://user-images.githubusercontent.com/2610395/82898780-c127cd80-9f51-11ea-81c4-3b72b7338a24.png)

**Comment by afharo** [comment]
> to that new version you mentioned

Oh! Sorry! Can you try with the latest `0.11.4`? I see other folks claiming it fixed it for them (only missing some minor bits #216)

**Comment by afharo** [comment]
Even better, `0.11.5` should also fix those minor bits :)

**Comment by robertybob** [comment]
I’m installing it as we speak 😃

**Comment by robertybob** [comment]
![0B87CC5D-531E-4A44-A2CB-80CFEBCFE040](https://user-images.githubusercontent.com/2610395/82932247-08c54e00-9f80-11ea-8bf6-3e4b59df415c.png)
![0DB39D02-6D54-460F-B981-A583F9313B52](https://user-images.githubusercontent.com/2610395/82932258-0c58d500-9f80-11ea-8daf-9208df6a0ee9.png)
![0A15AE7B-D127-4342-B96F-0C1B7C905063](https://user-images.githubusercontent.com/2610395/82932264-0d8a0200-9f80-11ea-8ef1-a75dcfdac790.png)

**Comment by afharo** [comment]
Oh... :(

It's kind of OK to see some errors at the beginning (I can work on that). But can you see if it reports OK after that? Or does it keep on failing to retrieve anything?

**Comment by robertybob** [comment]
![9FA08997-41A1-49D3-AE39-0CCAB27FF8A1](https://user-images.githubusercontent.com/2610395/82933564-019f3f80-9f82-11ea-81c0-e64b6619492b.png)
![AA2849B0-DE99-4E35-890A-8BD66086FFAB](https://user-images.githubusercontent.com/2610395/82933575-05cb5d00-9f82-11ea-8436-3c3166ec0f01.png)

**Comment by afharo** [comment]
Oh! Shoot! I just realised this is different to the Viomi model!

I can see in the python library there is a PR with added support to this model. I'll try to work on adding support to it based on those changes: https://github.com/rytilahti/python-miio/pull/683/files#diff-1d60d6cac3e7d3a87e527874901056af

Sorry for all the testing for nothing!

**Comment by robertybob** [comment]
> Oh! Shoot! I just realised this is different to the Viomi model!
> 
> I can see in the python library there is a PR with added support to this model. I'll try to work on adding support to it based on those changes: https://github.com/rytilahti/python-miio/pull/683/files#diff-1d60d6cac3e7d3a87e527874901056af
> 
> Sorry for all the testing for nothing!
 
No need to apologise; I’ve only been installing the code, you’ve been writing the code and doing all of the trouble shooting! I’m glad that the other model seems to be working fine now 👍 

That’s good news! I will keep my fingers crossed. But please, don’t stress about it too much if it isn’t possible to do, I appreciate all of the time and effort you’ve already spent on me.

**Comment by quartist** [comment]
Hello,
at this moment, 1C is partially supported or unsupported?

**Comment by robertybob** [comment]
> Hello,
> at this moment, 1C is partially supported or unsupported?

Not supported, but the developer hopes that it will be at some point in the future

**Comment by afharo** [comment]
Hey! Sorry for being dormant on this. I didn't have much time to implement it yet (I was also waiting on the linked Python project to move forward so we know the right commands we need to send to this model).

I've updated the description with some Developer's notes of the findings so far :)

**Comment by RaftTone** [comment]
Hey @afharo! I've just set up Homebridge for my IP cam and also was worried if I could add my Xiaomi 1C cleaner using this plugin so I've found this thread :) Would be great to have this model supported soon! Thanks so much for your work on this one!

**Comment by RaftTone** [comment]
Hey @afharo! Would be great to have this model supported soon. Thanks so much for your work on this one! I can help with tests & logs if you need some. Let me know!

**Comment by zibo29** [comment]
Good evening everybody, any follow up on this? thanks!

**Comment by ema07cqc** [comment]
Good morning everybody. any follow up on this? Dreame F9 user here. thanks!

**Comment by anisim101** [comment]
Does Xiaomi Mi Robot Vacuum Mop / mop 1c support?

**Comment by afharo** [comment]
> Hey! Sorry for being dormant on this. I didn't have much time to implement it yet (I was also waiting on the linked Python project to move forward so we know the right commands we need to send to this model).
> 
> I've updated the description with some Developer's notes of the findings so far :)

Hey everyone! Mentioning my last update. Unfortunately, adding a new model is not a straightforward task. Cases like this model imply reworking each one of the commands that the module needs to send to the robot. On top of that, I don't own that model, and I cannot test any attempts.

I'll try to come up with a guide for unsupported model owners to find and provide the set of commands for their model. Implementing them, once we know them all, should be a piece of cake.

**Comment by jeatrexs** [comment]
Hi, i'm new to this, and I'm also owner of the Xiaomi Mijia 1C model.
As i figured out by this thread this homebridge plugin is not able to communicate with that model yet.

Are there any news to this? Is there anything i could do to help, adding support for this model?

**Comment by MB-Jan** [comment]
+1

**Comment by MeichelCZ** [comment]
+1. I look forward to support 1C too.

**Comment by calvinngkk** [comment]
Yeah tell what we can do and we try to contribute

**Comment by MeichelCZ** [comment]
I can remotely offer my Raspberry Pi device with latest Hoobs installed, IP and token of vacuum cleaner Xiaomi 1C at my home. You can test what is needed and monitor the logs in your plugin.

**Comment by rafaelmagu** [comment]
Good news, everyone! Support for 1C has been merged into python-miio in https://github.com/rytilahti/python-miio/pull/952.

I _assume_ this will make it easier for this project to add the support for device as well, but I defer to @afharo to confirm this assumption.

**Comment by robertybob** [comment]
@rafaelmagu I don't want to get my hopes up too much, but I'm sure this is excellent news

**Comment by Treblame** [comment]
Hi.
I'm new here and I'm very interested in this thread.

I'm just wondering if there's a possible fix in these codes for the internal error in Xiaomi Mijia 1C, "Please check and clean my filter."

Thank you.

**Comment by RaftTone** [comment]
For everyone waiting I’d like to share a temporary solution: I got successfully running my 1C via Home Assistant and then added HA as a bridge to Homekit. Of course it has some problems & crashes that are fixed by rebooting HA instance all the time but at least it works to run & move back to the dock & show the charge of the vacuum.

![C8734701-3DE6-474C-8E77-F3E35226EEEA](https://user-images.githubusercontent.com/4950830/114638722-f6a5e880-9cd4-11eb-94f9-17bc98914012.png)

**Comment by anisim101** [comment]
> For everyone waiting I’d like to share a temporary solution: I got successfully running my 1C via Home Assistant and then added HA as a bridge to Homekit. Of course it has some problems & crashes that are fixed by rebooting HA instance all the time but at least it works to run & move back to the dock & show the charge of the vacuum.
> 
> 
> 
> ![C8734701-3DE6-474C-8E77-F3E35226EEEA](https://user-images.githubusercontent.com/4950830/114638722-f6a5e880-9cd4-11eb-94f9-17bc98914012.png)
Please give a link on tutorial

**Comment by RaftTone** [comment]
> Please give a link on tutorial

Instruction for the whole setup to work:
1. Have an IP & token for the 1C vacuum (the easiest way is to use [this app](https://github.com/Maxmudjon/Get_MiHome_devices_token) or use instructions in this repository)
2. Have Home Assistant installed
3. Tutorial on how to add the 1C vacuum to Home Assistant: [link](https://stupidhouse.info/node/34/)
It's in Russian (use translator in chrome). An important thing to add to that tutorial is that in configuration.yaml file 1 more line after 'token: Vacuum_Token' should be added:
version: 1.0.0
Otherwise the latest HA won't add a variable of the vacuum. P.S. HACS is not needed for the Homekit integration (skip it in the second part of the tutorial if that Lovelace card is not needed for you)
4. Add Homekit integration inside Home Assistant & link this virtual bridge to the Homekit app

As I mentioned before this setup is very unstable and I loose the vacuum in Homekit within several minutes and have to restart Home Assistant server to have it connected again. Still trying to figure this out.

**Comment by anisim101** [comment]
> > Please give a link on tutorial
> 
> 
> 
> Instruction for the whole setup to work:
> 
> 1. Have an IP & token for the 1C vacuum (the easiest way is to use [this app](https://github.com/Maxmudjon/Get_MiHome_devices_token) or use instructions in this repository)
> 
> 2. Have Home Assistant installed
> 
> 3. Tutorial on how to add the 1C vacuum to Home Assistant: [link](https://stupidhouse.info/node/34/)
> 
> It's in Russian (use translator in chrome). An important thing to add to that tutorial is that in configuration.yaml file 1 more line after 'token: Vacuum_Token' should be added:
> 
> version: 1.0.0
> 
> Otherwise the latest HA won't add a variable of the vacuum. P.S. HACS is not needed for the Homekit integration (skip it in the second part of the tutorial if that Lovelace card is not needed for you)
> 
> 4. Add Homekit integration inside Home Assistant & link this virtual bridge to the Homekit app
> 
> 
> 
> As I mentioned before this setup is very unstable and I loose the vacuum in Homekit (& the whole HA bridge connection) within several minutes and have to restart Home Assistant server (or just reload the Homekit integration in HA) to have it connected again.
> 
> As forums say it might be because if IGMP tick on or off in router settings or just the router has to be power cycled (didn't help me at all) or because of a hub (Apple TV) is connected wirelessly & goes to sleep after some mins and drops connection with HA.
> 
> Still trying to fix this out somehow. But at the moment it's not dropping connection for an hour or so so I can call it a little success :)
Спасибо :)

**Comment by McDin** [comment]
If you still search for a plug-in that supports the Dreame F9 or similar miot vacuum cleaner: https://github.com/merdok/homebridge-miot

**Comment by RaftTone** [comment]
> If you still search for a plug-in that supports the Dreame F9 or similar miot vacuum cleaner: https://github.com/merdok/homebridge-miot

Unfortunately it does not support dreame.vacuum.mc1808 which is 1C

**Comment by McDin** [comment]
Have you tested it? I have opened a thread here for my dreame and it was mentioned that the actions are similar to the Mija 1C

**Comment by RaftTone** [comment]
> Have you tested it? I have opened a thread here for my dreame and it was mentioned that the actions are similar to the Mija 1C

Will test this out, thanks for letting know!

**Comment by RaftTone** [comment]
> Have you tested it? I have opened a thread here for my dreame and it was mentioned that the actions are similar to the Mija 1C

Hey,
I've tried this one but it's not working. The plugin connects to the vacuum and shows it in HomeKit app as connected but it's not working. I've also tried specifying the device model dreame.vacuum.p2008 instead of mc1008 but it's not working either.

**Comment by McDin** [comment]
> > Have you tested it? I have opened a thread here for my dreame and it was mentioned that the actions are similar to the Mija 1C
> 
> Hey,
> I've tried this one but it's not working. The plugin connects to the vacuum and shows it in HomeKit app as connected but it's not working. I've also tried specifying the device model dreame.vacuum.p2008 instead of mc1008 but it's not working either.

Maybe the actions are others. You can ask there if he can implement it. We do it so for my vacuum robot. Some testing and the he implemented it

**Comment by RaftTone** [comment]
> > > Have you tested it? I have opened a thread here for my dreame and it was mentioned that the actions are similar to the Mija 1C
> > 
> > 
> > Hey,
> > I've tried this one but it's not working. The plugin connects to the vacuum and shows it in HomeKit app as connected but it's not working. I've also tried specifying the device model dreame.vacuum.p2008 instead of mc1008 but it's not working either.
> 
> Maybe the actions are others. You can ask there if he can implement it. We do it so for my vacuum robot. Some testing and the he implemented it

Thanks for the tip. I confirm that plug-in works now with 1C

**Comment by anisim101** [comment]
Please, send me a link on tutorial. > +1 The 1C works great with the homebridge-miot plugin.

**Comment by MB-Jan** [comment]
I simply followed the instructions of the plugin. No special setup or tutorial needed. If you are stuck, I recommend to open an issue directly in the homebridge-miot plugin repo.


Best regards,

Jan Rohweder
Managing Director
[Marketing Bear Logo]<https://marketingbear.com>
Prompan Tower 3, 8th Floor Room 803 - 804, No. 3, Soi Ladprao 3, Chom Phon, Chatuchack, Bangkok 10900
[Phone Icon] +66 (0) 2 024 6632 | [Mobile Phone Icon]  +66 (0) 8 823 9 5516
[Home Icon] WEBSITE<https://www.marketingbear.com/> | [Map Icon]  ***@***.***,100.5643605,20.78z/data=!4m12!1m6!3m5!1s0x30e2997d954f6eef:0x7b53e12fc42220!2sJune+The+Bear+Co.,+Ltd.!8m2!3d13.780683!4d100.491691!3m4!1s0x0:0xb91f7a1c3f16f8fc!8m2!3d13.8122211!4d100.5645347> | [Email Icon]  ***@***.***> | [LinkedIn Icon]  LINKEDIN<https://linkedin.com/in/jrohw>
Marketing Bear Availability Calendar: Click Here<https://calendar.google.com/calendar?cid=NGhoMzhuamRldG5lcnZ2YjZhMm41NGpxbHNAZ3JvdXAuY2FsZW5kYXIuZ29vZ2xlLmNvbQ>
Confidentiality Note: This email may contain confidential and/or private information. If you received this email in error please delete and notify sender.
________________________________

**Comment by fortil** [comment]
> +1 The 1C works great with the homebridge-miot plugin.

I followed the steps to download the info and send the [plugin](https://stupidhouse.info/node/34/) to that folder and after I downloaded `homebridge-miot` and I put all information with the folder of that and everything is working on my homekit.

**Comment by zibo29** [comment]
> > +1 The 1C works great with the homebridge-miot plugin.
> 
> I followed the steps to download the info and send the [plugin](https://stupidhouse.info/node/34/) to that folder and after I downloaded `homebridge-miot` and I put all information with the folder of that and everything is working on my homekit.

Can you help me on this? i would like to add Mi 1C to Homekit using Homebridge - thanks!
